### PR TITLE
Add occurrences to TimeInformation component

### LINKED ISSUE #3604: Improve 'see all occurrences' button
Improve the button to display all occurrences from booking details dialog

### PR DESCRIPTION
**Pull request by meluru**
closes #3604

### CODE REVIEW

File: indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx [hunk 1]
```diff
@@ -269,12 +277,11 @@ class BookRoomModal extends React.Component {
         if (!room) {
             return null;
         }
-
+        const occurrencesNumber = availability && availability.dateRange.length;
```
Comment (meluru) [inline indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: I'm not sure if that's the best way to count occurrences in this component. What do you think?

Comment (ThiefMaster) [reply indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: `numOccurrences` or `occurrenceCount`, the current name sounds like some number assigned to a set of occurrences.

Regarding the way to count them.. seems OK to me. At least I can't think of a better way right now.

File: indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx [hunk 2]
```diff
@@ -269,12 +277,11 @@ class BookRoomModal extends React.Component {
...
         const conflictsExist = availability && !!Object.keys(availability.conflicts).length;
         const bookingBlocked = ({submitting, submitSucceeded}) => submitting || submitSucceeded;
         const buttonsBlocked = (fprops) => bookingBlocked(fprops) || (conflictsExist && !skipConflicts);
         const {is_auto_confirm: isDirectlyBookable} = room;
-        const link = <a onClick={() => this.setState({bookingConflictsVisible: true})} />;
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: `onClick={this.showConflicts}`

File: indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx [hunk 3]
```diff
@@ -258,6 +258,14 @@ class BookRoomModal extends React.Component {
         onClose();
     };
 
+    showConflicts = () => {
+        this.setState({bookingConflictsVisible: true});
+    };
+
+    hideConflicts = () => {
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: you're not using this anywhere

File: indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx [hunk 4]
```diff
@@ -197,7 +197,7 @@ class BookRoomModal extends React.Component {
     }
 
     renderRoomTimeline(availability) {
-        const hourSeries = _.range(6, 24, 2);
+        const hourSeries = _.range(6, 22, 2);
```
Comment (mic4ael) [inline indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: What is the reason for this change?

Comment (meluru) [reply indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx]: your comment from my previous PR:
https://github.com/indico/indico/pull/3558#discussion_r221244115
